core/src/main/java/io/kestra/core/runners/Worker.java

```diff
@@ -607,6 +604,12 @@ public void close() throws Exception {
         metricEntryQueue.close();
     }
 
+    // Used for tests
+    public void killWorker() {
+        log.warn("Kill called, shutting down");
+        this.workerThreadReferences.forEach(WorkerThread::kill);
+    }
```

Comment:
I'm not comfortable to have this method only for testing, at least it must be annotated with `@VisibleForTesting` but it can lead to issues if someone uses it by inadvertence somewhere else.

Maybe this can be worked around by starting the worker in a separate thread and killing this thread in the test?